packages/ra-core/src/controller/useListContext.ts

```diff
@@ -0,0 +1,12 @@
+import { useContext } from 'react';
+
+import ListContext from './ListContext';
+
+/**
+ * Hook to read the list controller props.
+ *
+ * Must be used within a <ListContext.Provider>
+ */
+const useListContext = () => useContext(ListContext);
```

Comment:
I it must be used inside a `ListContext`, maybe we can log a useful console message about it if it's not ?
See https://kentcdodds.com/blog/how-to-use-react-context-effectively.

```suggestion
const useListContext = () => {
    const context = useContext(ListContext);
    if (context === undefined) {
        throw new Error('useCountState must be used within a <ListContext.Provider>')
    }
    return context;
}
```

Reply:
already done :)

Reply:
Sorry about that. Missed it while reviewing commit by commit. :+1: